internal/supervise/main.go

```diff
@@ -126,11 +127,25 @@ func Main() {
 
 	// Proxy logs.
 	syslogProcID := strconv.Itoa(child.Pid)
-	go pipeToSyslog(ctx, conn, cfg.ComponentID, "out", syslogProcID, stdout)
-	go pipeToSyslog(ctx, conn, cfg.ComponentID, "err", syslogProcID, stderr)
 
-	// Wait for child process to exit.
+	var wg sync.WaitGroup
+	work := func(f func()) {
```

Comment:
instead of introducing this `work` function, you can just do `wg.Add(2)` and leave the `go pipeToSyslog` calls alone.

Reply:
oh er nevermind, ignore me here